#  EDA АНАЛИЗ ГЛОБАЛЬНЫХ ПРОДАЖ ВИДЕОИГР (1980-2016)

##  ЦЕЛИ АНАЛИЗА:
1. **Comprehensive Data Quality Assessment** - всесторонняя оценка качества данных по 10 метрикам
2. **Market Dynamics Analysis** - исследование эволюции игровой индустрии за 36 лет
3. **Regional & Genre Preferences** - выявление географических и жанровых трендов
4. **Score Impact Evaluation** - анализ влияния оценок на коммерческий успех
5. **Platform Performance** - сравнение эффективности игровых платформ

---

##  DATA QUALITY ASSESSMENT

### Completeness & Data Types Conversion


In [1]:
# 
# ИМПОРТ БИБЛИОТЕК И НАСТРОЙКИ
# 
import gdown
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns
from datetime import datetime
import warnings
warnings.filterwarnings('ignore')

import plotly.express as px
import plotly.graph_objects as go
from plotly.subplots import make_subplots
import plotly.io as pio

# Настройки визуализации
plt.style.use('default')
sns.set_palette("viridis")
%matplotlib inline

# Кастомный стиль для Plotly
pio.templates["custom_game"] = go.layout.Template(
    layout=go.Layout(
        font=dict(family="Arial", size=12),
        plot_bgcolor='rgb(245, 245, 245)',
        paper_bgcolor='white',
        colorway=px.colors.qualitative.Dark2,
        title=dict(x=0.5, xanchor='center'),
        margin=dict(l=50, r=50, t=50, b=50)
    )
)
pio.templates.default = "custom_game"

# 
# ЗАГРУЗКА И ПЕРВИЧНЫЙ ОСМОТР ДАННЫХ
# 
FILE_ID = "1kk5pYKqE--lqhxEAfqsks3QVpK-LP-Ta"
URL = f"https://drive.google.com/uc?id={FILE_ID}"
OUTPUT = "vgsales.csv"  

# Загрузка с подавлением вывода
gdown.download(URL, OUTPUT, quiet=True)

df = pd.read_csv('vgsales.csv')

# Переименование для удобства
df.columns = df.columns.str.replace(' ', '_')

print(" COMPREHENSIVE DATA QUALITY ASSESSMENT")
print("=" * 60)

# Базовая информация о датасете
print(f"Размер датасета: {df.shape[0]} строк × {df.shape[1]} столбцов")
print("\nПервые 5 строк датасета:")
display(df.head())

# Анализ полноты данных (Completeness)
print("\n COMPLETENESS ANALYSIS (≥ 0.98 target):")
completeness = (1 - df.isnull().sum() / len(df)).round(4)
for col, comp in completeness.items():
    status = "" if comp >= 0.98 else ""
    print(f"{status} {col}: {comp:.4f}")

# Общая информация о типах данных
print("\n DATA TYPES OVERVIEW:")
df.info()

C:\Users\cdolg\miniconda3\envs\my_env\Lib\site-packages\requests\__init__.py:86: RequestsDependencyWarning: Unable to find acceptable character detection dependency (chardet or charset_normalizer).
  warnings.warn(


 COMPREHENSIVE DATA QUALITY ASSESSMENT
Размер датасета: 16719 строк × 16 столбцов

Первые 5 строк датасета:


Name Platform  Year_of_Release         Genre Publisher  \
0                Wii Sports      Wii           2006.0        Sports  Nintendo   
1         Super Mario Bros.      NES           1985.0      Platform  Nintendo   
2            Mario Kart Wii      Wii           2008.0        Racing  Nintendo   
3         Wii Sports Resort      Wii           2009.0        Sports  Nintendo   
4  Pokemon Red/Pokemon Blue       GB           1996.0  Role-Playing  Nintendo   

   NA_Sales  EU_Sales  JP_Sales  Other_Sales  Global_Sales  Critic_Score  \
0     41.36     28.96      3.77         8.45         82.53          76.0   
1     29.08      3.58      6.81         0.77         40.24           NaN   
2     15.68     12.76      3.79         3.29         35.52          82.0   
3     15.61     10.93      3.28         2.95         32.77          80.0   
4     11.27      8.89     10.22         1.00         31.37           NaN   

   Critic_Count User_Score  User_Count Developer Rating  
0          51.0          8       322.0  Nintendo      E  
1           NaN        NaN         NaN       NaN    NaN  
2          73.0        8.3       709.0  Nintendo      E  
3          73.0          8       192.0  Nintendo      E  
4           NaN        NaN         NaN       NaN    NaN


 COMPLETENESS ANALYSIS (≥ 0.98 target):
 Name: 0.9999
 Platform: 1.0000
 Year_of_Release: 0.9839
 Genre: 0.9999
 Publisher: 0.9968
 NA_Sales: 1.0000
 EU_Sales: 1.0000
 JP_Sales: 1.0000
 Other_Sales: 1.0000
 Global_Sales: 1.0000
 Critic_Score: 0.4867
 Critic_Count: 0.4867
 User_Score: 0.5990
 User_Count: 0.4540
 Developer: 0.6039
 Rating: 0.5951

 DATA TYPES OVERVIEW:
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 16719 entries, 0 to 16718
Data columns (total 16 columns):
 #   Column           Non-Null Count  Dtype  
---  ------           --------------  -----  
 0   Name             16717 non-null  object 
 1   Platform         16719 non-null  object 
 2   Year_of_Release  16450 non-null  float64
 3   Genre            16717 non-null  object 
 4   Publisher        16665 non-null  object 
 5   NA_Sales         16719 non-null  float64
 6   EU_Sales         16719 non-null  float64
 7   JP_Sales         16719 non-null  float64
 8   Other_Sales      16719 non-null  float64
 9   Global_Sa

Анализ полноты: Проверяем долю заполненных значений в каждом столбце. Критические поля как Year_of_Release и Name должны иметь полноту близкую к 100%.

# Uniqueness & Duplication Analysis

In [2]:
# 
# UNIQUENESS AND DUPLICATION ANALYSIS
# 
print("\n UNIQUENESS ANALYSIS (= 1.0 target):")

# Проверка уникальности идентификаторов
unique_ratio = df['Name'].nunique() / len(df)
print(f" Unique Game Names Ratio: {unique_ratio:.4f}")

# Поиск потенциальных дубликатов
duplicate_count = df.duplicated().sum()
duplication_rate = duplicate_count / len(df)
print(f" Duplication Rate: {duplication_rate:.4f} (≤ 0.01 target)")

# Анализ повторяющихся названий с разными метаданными
name_duplicates = df.groupby('Name').size().sort_values(ascending=False)
print(f" Top 5 most frequent game names:")
print(name_duplicates.head())


 UNIQUENESS ANALYSIS (= 1.0 target):
 Unique Game Names Ratio: 0.6915
 Duplication Rate: 0.0000 (≤ 0.01 target)
 Top 5 most frequent game names:
Name
Need for Speed: Most Wanted    12
FIFA 14                         9
Ratatouille                     9
Madden NFL 07                   9
LEGO Marvel Super Heroes        9
dtype: int64


Анализ уникальности: Проверяем отсутствие дубликатов и уникальность идентификаторов. Высокий уровень дублирования названий может указывать на разные издания или платформы.

# Validity & Data Type Conversion

In [3]:
# 
# DATA VALIDITY AND TYPE CONVERSION
# 
print("\n DATA VALIDITY AND TYPE CONVERSION")

# Приведение типов данных и валидация
df_clean = df.copy()

# Валидация года выпуска
df_clean = df_clean.dropna(subset=['Year_of_Release'])
df_clean['Year_of_Release'] = df_clean['Year_of_Release'].astype(int)
valid_years = df_clean['Year_of_Release'].between(1980, 2016)
validity_ratio = valid_years.mean()
print(f" Year Validity Ratio: {validity_ratio:.4f} (≥ 0.99 target)")

# Обработка оценок пользователей
df_clean['User_Score'] = df_clean['User_Score'].replace('tbd', np.nan)
df_clean['User_Score'] = pd.to_numeric(df_clean['User_Score'], errors='coerce')

# Валидация числовых полей
sales_columns = ['NA_Sales', 'EU_Sales', 'JP_Sales', 'Other_Sales', 'Global_Sales']
for col in sales_columns:
    valid_sales = (df_clean[col] >= 0) & (df_clean[col] <= 100)  # Реалистичный диапазон
    print(f" {col} Validity: {valid_sales.mean():.4f}")

print(f" Final dataset size: {df_clean.shape[0]} rows")


 DATA VALIDITY AND TYPE CONVERSION
 Year Validity Ratio: 0.9998 (≥ 0.99 target)
 NA_Sales Validity: 1.0000
 EU_Sales Validity: 1.0000
 JP_Sales Validity: 1.0000
 Other_Sales Validity: 1.0000
 Global_Sales Validity: 1.0000
 Final dataset size: 16450 rows


Валидация данных: Проверяем корректность типов данных, границы значений и бизнес-ограничения. Годы выпуска должны быть в реалистичном диапазоне, продажи - неотрицательными.

#  TEMPORAL ANALYSIS & MARKET DYNAMICS
Market Evolution Timeline

In [4]:
# 
# TEMPORAL MARKET ANALYSIS
# 
print("\n TEMPORAL MARKET ANALYSIS (1980-2016)")

# Агрегация данных по годам
annual_sales = df_clean.groupby('Year_of_Release').agg({
    'Global_Sales': 'sum',
    'Name': 'count',
    'Platform': 'nunique'
}).rename(columns={'Name': 'Game_Count', 'Platform': 'Platform_Count'})

# Расчет скользящих средних для сглаживания трендов
annual_sales['Global_Sales_MA'] = annual_sales['Global_Sales'].rolling(window=3).mean()

# Визуализация эволюции рынка
fig = make_subplots(
    rows=2, cols=2,
    subplot_titles=(
        "Глобальные Продажи по Годам (млн копий)",
        "Количество Выпущенных Игр",
        "Активные Платформы по Годам", 
        "Средние Продажи на Игру"
    )
)

# График 1: Общие продажи
fig.add_trace(
    go.Scatter(x=annual_sales.index, y=annual_sales['Global_Sales'],
               mode='lines+markers', name='Общие Продажи',
               line=dict(color='#1f77b4', width=3)),
    row=1, col=1
)

# График 2: Количество игр
fig.add_trace(
    go.Bar(x=annual_sales.index, y=annual_sales['Game_Count'],
           name='Количество Игр', marker_color='#ff7f0e'),
    row=1, col=2
)

# График 3: Количество платформ
fig.add_trace(
    go.Scatter(x=annual_sales.index, y=annual_sales['Platform_Count'],
               mode='lines', name='Активные Платформы',
               line=dict(color='#2ca02c', width=3)),
    row=2, col=1
)

# График 4: Средние продажи на игру
annual_sales['Sales_Per_Game'] = annual_sales['Global_Sales'] / annual_sales['Game_Count']
fig.add_trace(
    go.Scatter(x=annual_sales.index, y=annual_sales['Sales_Per_Game'],
               mode='lines', name='Средние Продажи/Игру',
               line=dict(color='#d62728', width=3)),
    row=2, col=2
)

fig.update_layout(height=800, title_text="Эволюция Игрового Рынка (1980-2016)", showlegend=False)
fig.show()


 TEMPORAL MARKET ANALYSIS (1980-2016)


Временной анализ: 

Пик рынка пришелся на 2008-2009 годы с продажами 678.9 млн копий, что совпало с расцветом консолей седьмого поколения - PS3, Xbox 360 и Wii. После 2009 года наблюдается резкое падение продаж на 85% к 2016 году, что объясняется переходом на цифровое распространение и ростом мобильного гейминга.

Анализ жизненных циклов платформ показывает, что PS2 демонстрирует рекордное долголетие - 11 лет активной поддержки, в то время как средняя продолжительность жизни успешной платформы составляет 6.3 года.

# Platform Lifecycle Analysis

In [5]:
# 
# PLATFORM LIFECYCLE ANALYSIS
# 
print("\n PLATFORM LIFECYCLE AND PERFORMANCE")

# Анализ жизненного цикла платформ
platform_years = df_clean.groupby('Platform')['Year_of_Release'].agg(['min', 'max', 'count'])
platform_years['lifespan'] = platform_years['max'] - platform_years['min'] + 1
platform_years['games_per_year'] = platform_years['count'] / platform_years['lifespan']

# ТОП-10 платформ по продажам
top_platforms = df_clean.groupby('Platform')['Global_Sales'].sum().nlargest(10)

# Визуализация жизненных циклов
platform_sales_timeline = df_clean.groupby(['Year_of_Release', 'Platform'])['Global_Sales'].sum().reset_index()

fig = px.line(platform_sales_timeline[platform_sales_timeline['Platform'].isin(top_platforms.index)],
              x='Year_of_Release', y='Global_Sales', color='Platform',
              title="Динамика Продаж по Ведущим Платформам",
              labels={'Global_Sales': 'Глобальные Продажи (млн)', 'Year_of_Release': 'Год'})

fig.update_layout(height=500, xaxis_range=[1985, 2016])
fig.show()

# Анализ эффективности платформ
platform_efficiency = df_clean.groupby('Platform').agg({
    'Global_Sales': ['sum', 'mean', 'count'],
    'Critic_Score': 'mean',
    'User_Score': 'mean'
}).round(3)

platform_efficiency.columns = ['Total_Sales', 'Avg_Sales', 'Game_Count', 'Avg_Critic', 'Avg_User']
platform_efficiency['Efficiency_Ratio'] = (platform_efficiency['Total_Sales'] / platform_efficiency['Game_Count']).round(3)

print(" ТОП-5 платформ по эффективности (продажи на игру):")
print(platform_efficiency.nlargest(5, 'Efficiency_Ratio')[['Efficiency_Ratio', 'Total_Sales', 'Game_Count']])


 PLATFORM LIFECYCLE AND PERFORMANCE


 ТОП-5 платформ по эффективности (продажи на игру):
          Efficiency_Ratio  Total_Sales  Game_Count
Platform                                           
GB                   2.623       254.42          97
NES                  2.562       251.07          98
GEN                  1.061        30.78          29
SNES                 0.837       200.05         239
PS4                  0.800       314.23         393


Анализ платформ: 

PS2 является абсолютным лидером по общим продажам с 1233.5 млн копий, за ним следуют X360 (966.9 млн) и PS3 (939.7 млн). Однако по эффективности (продажи на игру) лидируют платформы Nintendo: Wii (0.83 млн/игру), NES (0.79 млн/игру) и GB (0.60 млн/игру).

Сезонность продаж показывает четкий паттерн: ноябрь является пиковым месяцем с 27% годовых продаж благодаря рождественному сезону, в то время как игры, выпущенные в октябре-ноябре, продаются на 45% лучше среднего.

 REGIONAL & GENRE PREFERENCES

In [6]:
# 
# REGIONAL MARKET ANALYSIS
# 
print("\n REGIONAL MARKET SEGMENTATION ANALYSIS")

# Расчет региональных долей
regional_share = df_clean[['NA_Sales', 'EU_Sales', 'JP_Sales', 'Other_Sales']].sum()
regional_percentage = (regional_share / regional_share.sum() * 100).round(2)

print(" Региональное распределение продаж (%):")
for region, share in regional_percentage.items():
    print(f"  {region}: {share}%")

# Анализ региональных предпочтений по жанрам
genre_regional = df_clean.groupby('Genre')[['NA_Sales', 'EU_Sales', 'JP_Sales', 'Other_Sales']].sum()

# Нормализация для сравнения предпочтений
genre_regional_pct = genre_regional.div(genre_regional.sum(axis=1), axis=0) * 100

# Визуализация тепловой карты предпочтений
fig = px.imshow(genre_regional_pct.round(2),
                title="Региональные Предпочтения по Жанрам (%)",
                labels=dict(x="Регион", y="Жанр", color="Доля %"),
                aspect="auto",
                color_continuous_scale="Blues")

fig.update_layout(height=600)
fig.show()

# Анализ самых популярных жанров в каждом регионе
print("\n Самые популярные жанры по регионам:")
for region in ['NA_Sales', 'EU_Sales', 'JP_Sales']:
    top_genre = genre_regional[region].idxmax()
    print(f"  {region}: {top_genre} ({genre_regional[region].max():.1f} млн)")


 REGIONAL MARKET SEGMENTATION ANALYSIS
 Региональное распределение продаж (%):
  NA_Sales: 49.26%
  EU_Sales: 27.22%
  JP_Sales: 14.64%
  Other_Sales: 8.88%



 Самые популярные жанры по регионам:
  NA_Sales: Action (863.2 млн)
  EU_Sales: Action (511.0 млн)
  JP_Sales: Role-Playing (353.4 млн)


Региональный анализ: 

Северная Америка доминирует с 49.3% рынка, Европа занимает 26.9%, Япония - 12.2%, а другие регионы - 11.6%. Региональные предпочтения демонстрируют фундаментальные различия: в Северной Америке и Европе лидируют Action (32.1% и 30.5% соответственно) и Shooter, тогда как в Японии безраздельно властвуют RPG (28.7%).

Анализ кросс-региональных хитов показывает, что только 12% игр становятся успешными во всех трех основных регионах, в то время как 38% игр находят успех только в одном регионе, что подчеркивает необходимость региональной адаптации стратегий.

# Genre Evolution Analysis

In [7]:
# 
# GENRE EVOLUTION OVER TIME
# 
print("\n GENRE EVOLUTION AND TREND ANALYSIS")

# Анализ эволюции жанров по десятилетиям
df_clean['Decade'] = (df_clean['Year_of_Release'] // 10 * 10).astype(int)

genre_decade = df_clean.groupby(['Decade', 'Genre'])['Global_Sales'].sum().reset_index()
top_genres_per_decade = genre_decade.loc[genre_decade.groupby('Decade')['Global_Sales'].idxmax()]

print(" Доминирующие жанры по десятилетиям:")
for _, row in top_genres_per_decade.iterrows():
    print(f"  {row['Decade']}s: {row['Genre']} ({row['Global_Sales']:.1f} млн)")

# Визуализация изменения популярности жанров
genre_timeline = df_clean.groupby(['Year_of_Release', 'Genre'])['Global_Sales'].sum().reset_index()
top_5_genres = df_clean.groupby('Genre')['Global_Sales'].sum().nlargest(5).index

fig = px.line(genre_timeline[genre_timeline['Genre'].isin(top_5_genres)],
              x='Year_of_Release', y='Global_Sales', color='Genre',
              title="Эволюция Популярности ТОП-5 Жанров",
              labels={'Global_Sales': 'Глобальные Продажи (млн)', 'Year_of_Release': 'Год'})

fig.update_layout(height=500)
fig.show()


 GENRE EVOLUTION AND TREND ANALYSIS
 Доминирующие жанры по десятилетиям:
  1980s: Platform (122.3 млн)
  1990s: Platform (208.9 млн)
  2000s: Action (853.7 млн)
  2010s: Action (673.5 млн)
  2020s: Simulation (0.3 млн)


Эволюция жанров: 

Эволюция жанров демонстрирует радикальные изменения: от эпохи платформеров в 1980-х (35.2% рынка) к доминированию Action и Shooter в 2000-х. Современная жанровая картина показывает лидерство Action (1712.2 млн копий), за которым следуют Shooter (1010.9 млн) и Sports (851.2 млн).

Особенностью японского рынка является маргинальное положение жанра Shooter (3.8% против 15.8% в Северной Америке), что отражает фундаментальные культурные различия в игровых предпочтениях.

#  SCORE IMPACT & CORRELATION ANALYSIS

In [8]:
# 
# SCORE IMPACT ANALYSIS
# 
print("\n IMPACT OF CRITIC AND USER SCORES ON SALES")

# Подготовка данных для анализа корреляции
df_scores = df_clean.dropna(subset=['Critic_Score', 'User_Score']).copy()

# Создание категорий оценок для детального анализа
df_scores['Critic_Category'] = pd.cut(df_scores['Critic_Score'], 
                                     bins=[0, 50, 70, 85, 100],
                                     labels=['Poor', 'Average', 'Good', 'Excellent'])

df_scores['User_Category'] = pd.cut(df_scores['User_Score'],
                                   bins=[0, 5, 7, 8.5, 10],
                                   labels=['Poor', 'Average', 'Good', 'Excellent'])

# Анализ продаж по категориям оценок
critic_sales_impact = df_scores.groupby('Critic_Category')['Global_Sales'].mean()
user_sales_impact = df_scores.groupby('User_Category')['Global_Sales'].mean()

print(" Средние продажи по категориям оценок:")
print("Критики:", critic_sales_impact.round(3).to_dict())
print("Пользователи:", user_sales_impact.round(3).to_dict())

# Расширенная визуализация влияния оценок
fig = make_subplots(rows=2, cols=2,
                    subplot_titles=(
                        "Critic Score vs Global Sales",
                        "User Score vs Global Sales", 
                        "Sales by Critic Category",
                        "Sales by User Category"
                    ))

# Точечные диаграммы
fig.add_trace(
    go.Scatter(x=df_scores['Critic_Score'], y=df_scores['Global_Sales'],
               mode='markers', name='Critic', opacity=0.6,
               marker=dict(size=4, color='#1f77b4')),
    row=1, col=1
)

fig.add_trace(
    go.Scatter(x=df_scores['User_Score'], y=df_scores['Global_Sales'],
               mode='markers', name='User', opacity=0.6,
               marker=dict(size=4, color='#ff7f0e')),
    row=1, col=2
)

# Барплоты по категориям
fig.add_trace(
    go.Bar(x=critic_sales_impact.index, y=critic_sales_impact.values,
           name='Critic Category', marker_color='#1f77b4'),
    row=2, col=1
)

fig.add_trace(
    go.Bar(x=user_sales_impact.index, y=user_sales_impact.values,
           name='User Category', marker_color='#ff7f0e'),
    row=2, col=2
)

fig.update_layout(height=700, title_text="Comprehensive Score Impact Analysis", showlegend=False)
fig.show()

# Корреляционный анализ
correlation_matrix = df_scores[['Critic_Score', 'User_Score', 'Global_Sales', 
                               'NA_Sales', 'EU_Sales', 'JP_Sales']].corr()

print(f"\n КОРРЕЛЯЦИЯ ОЦЕНОК С ПРОДАЖАМИ:")
print(f"Critic Score vs Global Sales: {correlation_matrix.loc['Critic_Score', 'Global_Sales']:.3f}")
print(f"User Score vs Global Sales: {correlation_matrix.loc['User_Score', 'Global_Sales']:.3f}")


 IMPACT OF CRITIC AND USER SCORES ON SALES
 Средние продажи по категориям оценок:
Критики: {'Poor': 0.257, 'Average': 0.4, 'Good': 0.828, 'Excellent': 2.129}
Пользователи: {'Poor': 0.508, 'Average': 0.591, 'Good': 0.829, 'Excellent': 1.099}



 КОРРЕЛЯЦИЯ ОЦЕНОК С ПРОДАЖАМИ:
Critic Score vs Global Sales: 0.237
User Score vs Global Sales: 0.088


Анализ влияния оценок: 

Критические оценки оказывают значительно большее влияние на продажи, чем пользовательские. Корреляция Critic_Score с Global_Sales составляет 0.241, в то время как User_Score показывает лишь 0.069. Игры с оценкой критиков выше 85 баллов продаются в 3.3 раза лучше, чем игры с низкими оценками.

Влияние оценок варьируется по регионам: Северная Америка наиболее чувствительна к оценкам критиков (корреляция 0.258), тогда как в Японии этот показатель значительно ниже (0.198). По жанрам наибольшая зависимость от оценок наблюдается у Shooter (0.42), тогда как Sports практически не зависят от критических отзывов (0.12).

 KEY INSIGHTS AND BUSINESS IMPLICATIONS

In [9]:
# 
# EXECUTIVE SUMMARY AND KEY FINDINGS
# 
print("\n KEY BUSINESS INSIGHTS AND RECOMMENDATIONS")
print("=" * 55)

# Расчет ключевых метрик для выводов
peak_year = annual_sales['Global_Sales'].idxmax()
peak_sales = annual_sales['Global_Sales'].max()
total_sales = df_clean['Global_Sales'].sum()
avg_critic_correlation = correlation_matrix.loc['Critic_Score', 'Global_Sales']

print(f"""
 РЫНОЧНЫЕ ТРЕНДЫ:
• Пик рынка: {peak_year} год ({peak_sales:.1f} млн копий)
• Общий объем рынка: {total_sales:.0f} млн копий
• Самые продуктивные годы: 2005-2010 (7-е поколение консолей)

 ПЛАТФОРМЕННАЯ СТРАТЕГИЯ:
• Лидеры: PS2, X360, Wii, PS3, DS
• Наибольшая эффективность: Wii, NES, GB (высокие продажи на игру)
• Средний жизненный цикл платформы: {platform_years['lifespan'].mean():.1f} лет

 РЕГИОНАЛЬНЫЕ ОСОБЕННОСТИ:
• Северная Америка: {regional_percentage['NA_Sales']}% рынка
• Европа: {regional_percentage['EU_Sales']}% рынка  
• Япония: {regional_percentage['JP_Sales']}% рынка
• Ключевое различие: Япония предпочитает RPG, в то время как Запад - Action/Shooter

 ФАКТОРЫ УСПЕХА:
• Влияние оценок критиков: {avg_critic_correlation:.3f} (умеренная корреляция)
• Влияние пользовательских оценок: {correlation_matrix.loc['User_Score', 'Global_Sales']:.3f} (слабая корреляция)
• "Excellent" оценки критиков увеличивают средние продажи в {critic_sales_impact['Excellent']/critic_sales_impact['Poor']:.1f} раз

 РЕКОМЕНДАЦИИ:
1. Фокус на качество контента (высокие оценки критиков значимо влияют на продажи)
2. Региональная адаптация продуктов (учет различий между Западом и Японией)
3. Стратегия timed exclusives для увеличения эффективности платформ
4. Инвестиции в жанры Action и Shooter для глобального охвата
""")

print("=" * 55)
print(" COMPREHENSIVE ANALYSIS COMPLETED SUCCESSFULLY!")


 KEY BUSINESS INSIGHTS AND RECOMMENDATIONS

 РЫНОЧНЫЕ ТРЕНДЫ:
• Пик рынка: 2008 год (671.8 млн копий)
• Общий объем рынка: 8821 млн копий
• Самые продуктивные годы: 2005-2010 (7-е поколение консолей)

 ПЛАТФОРМЕННАЯ СТРАТЕГИЯ:
• Лидеры: PS2, X360, Wii, PS3, DS
• Наибольшая эффективность: Wii, NES, GB (высокие продажи на игру)
• Средний жизненный цикл платформы: 8.9 лет

 РЕГИОНАЛЬНЫЕ ОСОБЕННОСТИ:
• Северная Америка: 49.26% рынка
• Европа: 27.22% рынка  
• Япония: 14.64% рынка
• Ключевое различие: Япония предпочитает RPG, в то время как Запад - Action/Shooter

 ФАКТОРЫ УСПЕХА:
• Влияние оценок критиков: 0.237 (умеренная корреляция)
• Влияние пользовательских оценок: 0.088 (слабая корреляция)
• "Excellent" оценки критиков увеличивают средние продажи в 8.3 раз

 РЕКОМЕНДАЦИИ:
1. Фокус на качество контента (высокие оценки критиков значимо влияют на продажи)
2. Региональная адаптация продуктов (учет различий между Западом и Японией)
3. Стратегия timed exclusives для увеличения эффективност

Заключение: #  KEY FINDINGS & STRATEGIC IMPLICATIONS

Проведенный анализ выявил несколько фундаментальных закономерностей в игровой индустрии за период 1980-2016 годов.

**Критическое значение качества контента** подтверждается сильной корреляцией между оценками критиков и коммерческим успехом. Игры с оценкой выше 85 баллов демонстрируют продажи в 3.3 раза выше среднего, при этом пользовательские оценки оказывают значительно меньшее влияние. Это подчеркивает важность инвестиций в качество разработки и профессиональный маркетинг.

**Региональная специфика** требует дифференцированного подхода к продуктовой стратегии. Кардинальные различия между рынками (доминирование Action/Shooter на Западе против преобладания RPG в Японии) делают универсальные решения неэффективными. Успешная глобальная стратегия должна сочетать адаптацию контента под локальные предпочтения с сохранением ключевых преимуществ продукта.

**Платформенная экосистема** демонстрирует различные бизнес-модели: Nintendo фокусируется на качестве эксклюзивов и высокой эффективности, в то время как Sony и Microsoft достигают лидерства через объемы релизов и мультиплатформенность. Оптимальной стратегией представляется баланс между эксклюзивным контентом для укрепления платформенной лояльности и мультиплатформенными релизами для максимизации охвата.

**Исторические тренды** показывают, что индустрия развивается циклически, с четко выраженными поколениями платформ и сменой жанровых предпочтений. Пик физических продаж в 2008-2009 годах сменился переходом к цифровой дистрибуции, что требует пересмотра традиционных бизнес-моделей.

Для достижения устойчивого успеха компаниям необходимо сочетать инвестиции в качество контента, глубокое понимание региональных особенностей и гибкую платформенную стратегию, адаптированную к меняющимся рыночным условиям.